test/common.py

```diff
@@ -1197,6 +1197,9 @@ def require_simd(self):
       self.fail('either d8 or node >= 16 required to run wasm64 tests.  Use EMTEST_SKIP_SIMD to skip')
 
   def require_wasm_legacy_eh(self):
+    if 'EMTEST_SKIP_WASM_LEGACY_EH' in os.environ:
+      self.skipTest('test requires node >= 17 or d8 (and EMTEST_SKIP_WASM_LEGACY_EH is set)')
```

Comment:
@aheejin now I realize my root source of the confusion. What was happening is that the semantics of `EMTEST_SKIP_EH=1` was to not actually skip the test, if the node.js shell was detected to support the target EH.

I would like to change the semantics to mean that if user passes `EMTEST_SKIP_EH=1` (or `EMTEST_SKIP_WASM_LEGACY_EH=1`), then do skip the tests, independent of Node.js version. (the tests are running in the browser, so Node.js version does not matter)

So what was happening all the time on my CI tests was that I was passing `EMTEST_SKIP_EH=1` and `EMTEST_SKIP_WASM_LEGACY_EH=1` and running in Firefox 91, which doesn't support either; but the tests would not be skipped because I am running on Node.js 24, and they would then proceed to fail in the old browser.

Reply:
I agree that's confusing. I actually wasn't aware that we don't skip the tests even if we have `EMTEST_SKIP_***` set if the node version is high enough. But it looks this has been this way for years and it doesn't only apply to Wasm EH. Other methods like `require_simd` or `require_wasm64` have the same problem. I think it would be better to change it in a separate PR for all `require_***`s if we are to change this. @sbc100 What do you think about changing the meaning of `EMTEST_SKIP_***`? Is there a reason that we don't honor these skip variables when the node version is high?